diskutils/diskutils.go

```diff
@@ -0,0 +1,110 @@
+package diskutils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+)
+
+// CreateTMPDIR creates tmp dir at path configured via RUDDER_TMPDIR env var
+func CreateTMPDIR() (string, error) {
+	tmpdirPath := strings.TrimSuffix(config.GetString("RUDDER_TMPDIR", ""), "/")
```

Comment:
1. This shouldn't use the global config variable
2. Why is it a useful function to deserve being in rudder-go-kit?

Reply:
https://github.com/rudderlabs/rudder-server/pull/3417#discussion_r1210016732

It is based on this conversation

Reply:
`RUDDER_TMPDIR` appears to be a rudder-server specific thing though... we should only use rudder-go-kit for common things

Reply:
@fracasula @lvrach Are we fine with having this function in rudder-server?

Reply:
Sure maybe move it in a `fsutil` package in RS instead of having it in `misc`.